hvplot/converter.py

```diff
@@ -671,6 +671,9 @@ def _process_data(self, kind, data, x, y, by, groupby, row, col,
             data = data.to_frame()
         if is_intake(data):
             data = process_intake(data, use_dask or persist)
+        if isinstance(getattr(data, "columns", None), pd.RangeIndex):
+            data = data.rename(columns=str)
```

Comment:
@Hoxbro the thing I was not sure with this change is that doesn't this line actually create a copy of the data? `self.source_data` is supposed to be a reference to the original data.

The approach I've implemented locally is to temporarily replace the columns of the source data with columns that contain string objects, in a try/finally block.

The way I was going may just be overkill!

Reply:
Yes. It makes a copy. Looking at the previous if-statements, I would not say `self.source_data` refers to the original data, e.g., a series is converted to a dataframe. I see `self.data_source` as a reference to the original data, though I have not taken a thorough look at the rest of the codebase to see if this is the case.  

https://github.com/holoviz/hvplot/blob/eed0e877eb9a848acdc3e31bd197ed5c50f8bfd5/hvplot/converter.py#L668-L677

Reply:
`self.data_source` is unused. We don't get any warning for that as flake is set up to basically ignore everything. Something else to fix (I'll ask Philipp if I can run `black` once).